monai/engines/evaluator.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+from monai.inferers.inferer import SimpleInferer
+from .workflow import Workflow
+from .utils import default_prepare_batch
+from .utils import CommonKeys as Keys
+
+
+class Evaluator(Workflow):
+    """
+    Base class for all kinds of evaluators, inherits from Workflow.
+
+    Args:
+        device (torch.device): an object representing the device on which to run.
+        val_data_loader (torch.DataLoader): Ignite engine use data_loader to run, must be torch.DataLoader.
+        prepare_batch (Callable): function to parse image and label for current iteration.
+        iteration_update (Callable): the callable function for every iteration, expect to accept `engine`
+            and `batchdata` as input parameters. if not provided, use `self._iteration()` instead.
+        key_val_metric (ignite.metric): compute metric when every iteration completed, and save average value to
+            engine.state.metrics when epoch completed. key_val_metric is the main metric to comapre and save the
+            checkpoint into files.
+        additional_metrics (dict): more Ignite metrics that also attach to Ignite Engine.
+        val_handlers (list): every handler is a set of Ignite Event-Handlers, must have `attach` function, like:
+            CheckpointHandler, StatsHandler, SegmentationSaver, etc.
+
+    """
+
+    def __init__(
+        self,
+        device,
+        val_data_loader,
+        prepare_batch=default_prepare_batch,
+        iteration_update=None,
+        key_val_metric=None,
+        additional_metrics=None,
+        val_handlers=None,
+    ):
+        super().__init__(
+            device,
+            1,
+            False,
+            val_data_loader,
```

Comment:
could use `device=device, max_epochs=1, amp=False, ... ` for code readability (same for the trainers)